api/nodes/views.py

```diff
@@ -1156,6 +1161,104 @@ def get_object(self):
         return fobj
 
 
+class NodeGroupsList(JSONAPIBaseView, generics.ListCreateAPIView, NodeMixin, ListFilterMixin, OSFGroupMixin):
+    """ The documentation for this endpoint can be found [here](https://developer.osf.io/#operation/nodes_groups_list)
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        AdminOrPublic,
+        base_permissions.TokenHasScope,
+    )
+    model_class = OSFGroup
+
+    serializer_class = NodeGroupsSerializer
+
+    required_read_scopes = [CoreScopes.NODE_OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.NODE_OSF_GROUPS_WRITE]
+    view_category = 'nodes'
+    view_name = 'node-groups'
+
+    def get_default_queryset(self):
+        return self.get_node().osf_groups
+
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    # overrides FilterMixin
+    def build_query_from_field(self, field_name, operation):
+        if field_name == 'permission':
+            node = self.get_node()
+            try:
+                groups_with_perm_ids = node.get_osf_groups_with_perms(operation['value']).values_list('id', flat=True)
+            except ValueError:
+                raise ValidationError('{} is not a filterable permission.'.format(operation['value']))
+            return Q(id__in=groups_with_perm_ids)
+
+        return super(NodeGroupsList, self).build_query_from_field(field_name, operation)
+
+    # overrides ListCreateAPIView
+    def get_serializer_class(self):
+        if self.request.method == 'POST':
+            return NodeGroupsCreateSerializer
+        else:
+            return NodeGroupsSerializer
+
+    # overrides ListCreateAPIView
+    def get_serializer_context(self):
+        """
+        Extra context for NodeGroupsSerializer
+        """
+        context = super(NodeGroupsList, self).get_serializer_context()
+        context['node'] = self.get_node(check_object_permissions=False)
+        return context
+
+
+class NodeGroupsDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, NodeMixin, OSFGroupMixin):
```

Comment:
Sure, but besides the base classes, a List and a Detail endpoint aren't going to share a lot of the same methods.  I DRY them up when it's a significant amount of shared code, like the `OSFGroupMemberBaseView` in this same PR.  But it doesn't hurt to dry them up here, I just like to be careful about the tradeoffs between being DRY and being readable.

Reply:
I think it's nice to share a base class in a lot of these because it makes it really obvious how the sub classes static members and inherited classes are different hence readability as a opposed to just DRY-ness, but these could go either way.